Elena_Poleschuk/Module 3 -  Java/Week 1/Hometask3/src/by/epam/tat/lecture1/calculator/Calculator.java

```diff
@@ -0,0 +1,46 @@
+package by.epam.tat.lecture1.calculator;
+
+public class Calculator {
+	
+	public void calculation() {
+		int result;
+		String action;
+		System.out.println("Calculator works with integer digits and simple actions: -, +, /, *.");
+		do {
+			result = Communicator.getValue();
+		//	Communicator.in.nextLine();  // Consume newline left-over
+			action = Communicator.getAction();
+			System.out.println(action);
+			while ( action.equals("=") != true){
+			try{
+				switch (action)
+					{
+						case "+":
+							result = (result + Communicator.getValue());
+							break;
+						case "-":
+							result = result - Communicator.getValue(); 
+							break;			
+						case "*":
+							result = result * Communicator.getValue(); 
+							break;	
+						case "/":
+							result = result / Communicator.getValue();
+							break;		
+						default:
+							System.out.print("Error: action is incorrect. ");
+					}
```

Comment:
I think you should break the loop cycle in this place. Because if action is incorrect you print result =0